vapi/simulator/simulator.go

```diff
@@ -1443,6 +1524,10 @@ func (s *handler) libraryItemUpdateSessionFileID(w http.ResponseWriter, r *http.
 		}
 		OK(w, ids)
 	case "remove":
+		if up.State != "ACTIVE" {
+			s.error(w, fmt.Errorf("removeFile not allowed in state %s", up.State))
+			return
+		}
 		delete(s.Update, id)
 		OK(w)
 	case "validate":
```

Comment:
`validate` impl would be also needed for complete match of the [vSphere API](https://developer.vmware.com/apis/vsphere-automation/latest/content/api/content/library/item/update-session/update_session_id/fileactionvalidate/post/).

Reply:
Ah I missed validate listed in the issue. I will add that, may not be till tomorrow though.

Reply:
Added client side API for validate. vcsim side still doesn't do any validation of files, but at least returns a response body now (`{has_errors=false}`). I can still add that, but client-side is ready to use

Reply:
Cool, thanks!